### PR TITLE
ARMA remove FourierOrder

### LINKED ISSUE #909: Removing FourierOrder from ARMA
--------
Issue Description
--------
While initially a useful construct, the burden imposed on both the user and developer to maintain the FourierOrder option for subdividing Fourier periods in the ARMA is significant. Many new users are confused or misunderstand what the node does, and it hides the clarity that comes from explicitly listing desired periods. Within the code, significant syntax is employed to support the structure, when really a single list of periods would be mathematically equivalent.

Recommendation: remove FourierOrders and simple use Fourier to describe the Fourier periods in the ARMA.

Note this task will also vastly simplify the multi-fidelity segmented training approach to the ARMA.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #909

##### What are the significant changes in functionality due to this change request?
Removes the FourierOrder node in the ARMA in favor of using the Fourier node.

A conversion script is included, and an email to the user list will be required.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
The tests have all passed, this is ready for review (and a predecessor to the cluster  ROM MR).

File: framework/SupervisedLearning/ARMA.py [hunk 1]
```diff
@@ -752,149 +731,61 @@ def _trainCDF(self,data):
...
+    #                 3:   cos(2pi*t/period[1]), ...
     fourierEngine = linear_model.LinearRegression(normalize=False)
 
     # if using zero-filter, cut the parts of the Fourier and values that correspond to the zero-value portions
     if zeroFilter:
       values = values[self.zeroFilterMask]
-      fourierEvaluations = dict((period,vals[self.zeroFilterMask]) for period,vals in fourierSeriesOriginal.items())
+      # TODO is this right?
+      fourierSignals = fourierSignalsFull[self.zeroFilterMask, :]
+      # OLD fourierSignals = dict((period, vals[self.zeroFilterMask]) for period, vals in fourierSeriesOriginal.items())
```
Comment (wangcj05) [inline framework/SupervisedLearning/ARMA.py]: It seems to me you need to fix the "TODO" here.

**Comment by PaulTalbot-INL** [comment]
Addressed comments (removed commented lines, and added in conversion script)

**Comment by PaulTalbot-INL** [comment]
User email:
```
RAVEN Users:

As of MR #910, we've simplified the input of the ARMA once again. This time, to decrease the chance of misunderstanding, we've removed the "FourierOrder" node, which was an option to subidivide the Fourier periods selected. Now, each Fourier period desired should be listed explicitly in the "Fourier" node.

To ease transition, a conversion script has been placed in raven/scripts/conversionScripts/ARMA_remove_fourier_orders.py

Thank you!

- RAVEN developers
```

**Comment by wangcj05** [comment]
@PaulTalbot-INL Windows test failed.

**Comment by PaulTalbot-INL** [comment]
Tests are passed with the new j4 l4 setup.

**Comment by wangcj05** [comment]
Manual got updated, tests passed, and conversion script is provided. 
PR Checklist is satisfied, and this PR can be merged.

**Comment by wangcj05** [comment]
@PaulTalbot-INL You can send out the email now.

**Pull request merged**